stdpopsim/slim_engine.py

```diff
@@ -948,17 +948,58 @@ def fix_time(event):
     condition_on_allele_frequency = []
     op_id = stdpopsim.ext.ConditionOnAlleleFrequency.op_id
     slim_mutation_ids = []
-    for x in _dfe_to_mtypes(contig).values():
+    slim_mutation_dfe = []
+    for i, x in _dfe_to_mtypes(contig).items():
         slim_mutation_ids.extend([u[0] for u in x])
+        slim_mutation_dfe.extend([i for u in x])
     for ee in extended_events:
-        if hasattr(ee, "mutation_type_id"):
-            mt_id = getattr(ee, "mutation_type_id")
+        if hasattr(ee, "single_site_id"):
+            mut_id = getattr(ee, "single_site_id")
```

Comment:
Can we not call this `mut_id`, since it's not a mutation ID (it's a DFE ID); maybe `single_site_id`? It's so hard for me to keep this straight already.